edgehog-device-runtime-containers/src/service/resource.rs

```diff
@@ -18,43 +18,83 @@
 
 use std::fmt::Debug;
 
-use astarte_device_sdk::Client;
-use tracing::instrument;
+use tracing::{debug, info, instrument};
 
 use super::{Id, Result};
 
-use crate::Docker;
+use crate::{
+    image::Image, properties::image::AvailableImage, properties::AvailableProp, properties::Client,
+    store::StateStore, Docker,
+};
 
 /// A resource in the nodes struct.
+#[allow(dead_code)]
 pub(crate) trait Resource: Into<NodeType> {
     fn dependencies(&self) -> Result<Vec<String>>;
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub(crate) enum NodeType {}
+pub(crate) enum NodeType {
+    Image(Image<String>),
+}
 
 impl NodeType {
     #[instrument(skip_all)]
-    pub(super) async fn store<D>(&mut self, _id: &Id, _device: &D) -> Result<()>
+    pub(super) async fn store<D>(
+        &mut self,
+        id: &Id,
+        store: &mut StateStore,
+        device: &D,
+    ) -> Result<()>
     where
-        D: Debug + Client + Sync,
+        D: Client + Sync + 'static,
     {
-        unimplemented!()
+        match &self {
+            NodeType::Image(image) => {
+                store.append(id, image.into()).await?;
+
+                AvailableImage::new(id, false).send(device).await?;
+
+                info!("stored image with id {}", id);
+            }
+        }
+
+        Ok(())
     }
 
     #[instrument(skip_all)]
-    pub(super) async fn create<D>(&mut self, _id: &Id, _device: &D, _client: &Docker) -> Result<()>
+    pub(super) async fn create<D>(&mut self, id: &Id, device: &D, client: &Docker) -> Result<()>
     where
-        D: Debug + Client + Sync,
+        D: Debug + Client + Sync + 'static,
     {
-        unimplemented!()
+        match self {
+            NodeType::Image(image) => {
+                image.inspect_or_create(client).await?;
```

Comment:
It seems this is the only place where `inspect_or_create` is called, but the boolean value is not used (even though I understand it could be useful to differentiate the case in which the Image already exists or not). Is it really  necessary to return Result<bool> rather than Result<()> at the moment?

Reply:
I've wanted to use it in a test, now I've added it.